apps/backend/src/app/modules/session/run/run-session.service.ts

```diff
@@ -251,12 +251,25 @@ export class RunSessionService {
 
     const replayHash = FileStoreService.getHashForBuffer(replayBuffer);
 
+    const tasks: [
+      ReturnType<typeof this.updateLeaderboards>,
+      ...Array<PromiseLike<any>>
+    ] = [
+      this.updateLeaderboards(tx, submittedRun, isPB, existingRun, replayHash)
+    ];
+
+    if (isPB)
+      tasks.push(
+        this.updateReplayFiles(
+          replayBuffer,
+          replayHash,
+          existingRun?.replayHash
+        )
+      );
+
     // We have two quite expensive, independent operations here, including a
     // file store. So we may as well run in parallel and await them both.
-    const [{ run, xpGain, isWR }] = await parallel(
-      this.updateLeaderboards(tx, submittedRun, isPB, existingRun, replayHash),
-      this.updateReplayFiles(replayBuffer, replayHash, existingRun?.replayHash)
-    );
+    const [{ run, xpGain, isWR }] = await parallel(...tasks);
```

Comment:
Honestly, my parallel thing is just dumb and should probably phase out - you can just do Promise.all(tasks)